floyd/client/files.py

```diff
@@ -6,70 +6,79 @@
 from floyd.log import logger as floyd_logger
 
 
-def get_files_in_directory(path, file_type):
+def get_unignored_file_paths(path, ignore_list=[], whitelist=[]):
     """
-    Gets the list of files in the directory and subdirectories
-    Respects .floydignore file if present
+    Given an ignore_list and a whitelist of glob patterns, returns the list of
+    file paths in a directory and its subdirectories that are not ignored.
     """
-    local_files = []
-    separator = os.path.sep
-    ignore_list = FloydIgnoreManager.get_list()
-
-    # make sure that subdirectories are also excluded
-    ignore_list_expanded = ignore_list + ["{}/**".format(item) for item in ignore_list]
-    floyd_logger.debug("Ignoring list : {}".format(ignore_list))
-    total_file_size = 0
+    unignored_files = []
 
     for root, dirs, files in os.walk(path):
         floyd_logger.debug("Root:{}, Dirs:{}".format(root, dirs))
-        ignore_dir = False
-        normalized_path = normalize_path(path, root)
-        for item in ignore_list_expanded:
-            if PurePath(normalized_path).match(item):
-                ignore_dir = True
-                break
-
-        if ignore_dir:
-            # Reset dirs to avoid going further down this directory
+
+        if ignore_path(unix_stle_path(root), ignore_list, whitelist):
+            # Reset dirs to avoid going further down this directory.
+            # Then continue to the next iteration of os.walk, which causes
+            # everything in this directory to be ignored.
             dirs[:] = []
             floyd_logger.debug("Ignoring directory : {}".format(root))
```

Comment:
Nevermind! I see a test case for this. It might be useful to put that comment here!